lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -1330,9 +1330,10 @@ bool TypeLoweringVisitor::visitExpr(SubaccessOp op) {
     return true;
   }
 
-  // Check for constant instances
-  if (ConstantOp arg =
-          dyn_cast_or_null<ConstantOp>(op.getIndex().getDefiningOp())) {
+  // Check for constant instances.  Only use this path if the constant fits in
+  // the subaccess.
+  ConstantOp arg = dyn_cast_or_null<ConstantOp>(op.getIndex().getDefiningOp());
+  if (arg && arg.getValue().getZExtValue() < vType.getNumElements()) {
```

Comment:
Hyper-nit: The code guarded by this condition (directly below) uses `.getExtValue()`, seems like these should be the same (although maybe known to not matter either way)?

Looks like the `inferReturnTypes` uses zext, so maybe that's best here too?